e2e/cypress/integration/integrations/zapier/oauth_spec.js

```diff
@@ -0,0 +1,440 @@
+                    // * Token should not be obscured
+                    cy.contains('*******************').should('not.exist');
+
+                    // * Copy button should exist
+                    cy.get('.fa-copy').should('exist');
+                });
+            });
+        });
+    });
+
+    it('MM-T648 Part 1 - OAuth 2.0 Zapier tests', () => {
+        cy.apiLogin(user1);
+        cy.visit(testChannelUrl1);
+
+        // # Navigate to OAuthApps in integrations menu
+        cy.get('#headerInfo').click();
+        cy.get('#integrations').click();
+        cy.get('#oauthApps').click();
+
+        // # Click on the Add button
+        cy.get('#addOauthApp').click();
+
+        // # Fill all fields
+        cy.get('#name').type('Test');
+        cy.get('#description').type('Test');
+        cy.get('#homepage').type('https://www.test.com/');
+        cy.get('#callbackUrls').type(getWebhookBaseUrl() + '/complete_oauth');
+
+        // # Save
+        cy.get('#saveOauthApp').click();
+
+        // * Copy button should be visible
+        cy.get('.fa-copy').should('exist');
+
+        // # Store client ID
+        cy.findByText('Client ID').parent().invoke('text').then((text) => {
+            cy.wrap(text.substring(11)).as('clientID');
+        });
+
+        // # Store client secret
+        cy.findByText('Client Secret').parent().invoke('text').then((text) => {
+            cy.wrap(text.substring(15)).as('clientSecret');
+        });
+
+        cy.get('@clientID').then((clientID) => {
+            oauthClientID = clientID;
+            cy.get('@clientSecret').then((clientSecret) => {
+                oauthClientSecret = clientSecret;
+
+                // # Send credentials
+                cy.postIncomingWebhook({url: getWebhookBaseUrl() + '/send_oauth_credentials', data: {appID: clientID, appSecret: clientSecret}});
+            });
+        });
+
+        // # Click Done
+        cy.get('#doneButton').click();
+    });
+
+    it('Aux - Exchange tokens', () => {
+        cy.apiLogin(user1);
+
+        // # Visit the webhook url to start the OAuth handshake
+        cy.visit(getWebhookBaseUrl() + '/start_oauth');
+
+        // # Click on the allow button
+        cy.findByText('Allow').click();
+
+        // * Exchange succesful
+        cy.findByText('OK').should('exist');
+    });
+
+    it('MM-T648 Part 2 - OAuth 2.0 Zapier tests', () => {
+        // # Visit a channel
+        cy.visit(testChannelUrl1);
+
+        // # Wait for it to load
+        cy.wait(TIMEOUTS.FIVE_SEC);
+        cy.getCurrentChannelId().then((channelId) => {
+            const message = 'OAuth test 01';
+
+            // # Post message using OAuth credentials
+            cy.postIncomingWebhook({url: getWebhookBaseUrl() + '/postOAuthMessage', data: {channelId, message}});
+
+            // * The message should be posted
+            cy.findByText(message).should('exist');
+        });
+    });
+
+    it('MM-T649 Edit Zapier', () => {
```

Comment:
We're not actually testing against Zapier so I suggest to change the name, maybe like general term of "OAuth 2.0 Application"